### PR TITLE
Refactoring SearchViews re: SPC/SearchView/ContralsAndResults allowing more flexibility

### PR DESCRIPTION
**Pull request by alexkb0009**
These edits on SPC-side are a precursor to creating the FilterBlocks UI wherein support for a `props.aboveTableComponent` & `props.aboveFacetListComponent` will be required. Code should be mostly self-explanatory and be "cleanup".

Functionally, everything should work exactly as it did before, but code should be somewhat cleaner hopefully.

Please look at alongside https://github.com/4dn-dcic/shared-portal-components/pull/44

### CODE REVIEW

File: src/encoded/static/components/browse/FileSearchView.js [hunk 1]
```diff
@@ -5,59 +5,38 @@ import PropTypes from 'prop-types';
...
 function FileTableWithSelectedFilesCheckboxes(props){
     const {
         // Common high-level props from Redux, or App.js, or App.js > BodyElement:
         context, href, schemas, navigate: propNavigate,
         windowHeight, windowWidth, registerWindowOnScrollHandler,
         toggleFullScreen, isFullscreen, session,
-        facets,
-        selectedFiles, selectFile, unselectFile, resetSelectedFiles,
-        columnExtensionMap
+        selectedFiles, selectFile, unselectFile, resetSelectedFiles, selectedFilesUniqueCount,
```
Comment (j1z0) [inline src/encoded/static/components/browse/FileSearchView.js:37]: Codacy found an issue: ['resetSelectedFiles' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Clones removed
==============
+ src/encoded/static/components/browse/BrowseView.js  -1
+ src/encoded/static/components/browse/FileSearchView.js  -3
+ src/encoded/static/components/browse/MicroscopySearchView.js  -3
+ src/encoded/static/components/browse/PublicationSearchView.js  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004&bid=20599883)

File: src/encoded/static/components/browse/BrowseView.js [hunk 2]
```diff
@@ -494,33 +316,26 @@ function BrowseTableWithSelectedFilesCheckboxes(props){
         windowHeight, windowWidth, registerWindowOnScrollHandler,
         toggleFullScreen, isFullscreen, session,
 
-        // Props from BrowseView:
-        facets,
-
         // Props from SelectedFilesController (separate from SPC SearchView SelectedItemsController):
-        selectedFiles, selectFile, unselectFile, resetSelectedFiles,
+        selectedFiles, selectFile, unselectFile, resetSelectedFiles, selectedFilesUniqueCount,
```
Comment (j1z0) [inline src/encoded/static/components/browse/BrowseView.js:320]: Codacy found an issue: ['resetSelectedFiles' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/BrowseView.js [hunk 3]
```diff
@@ -537,40 +352,101 @@ function BrowseTableWithSelectedFilesCheckboxes(props){
...
+    const detailPane = useMemo(function(){
+        return (
+            <ExperimentSetDetailPane {...{ selectedFiles, selectFile, unselectFile, windowWidth, href, schemas }} paddingWidth={47}
+                initialStateCache={detailPaneFileSectionStateCache} updateFileSectionStateCache={updateDetailPaneFileSectionStateCache} />
+        );
+    }, [ selectedFiles, windowWidth, href, schemas ]);
+
+    /** Some data calculated/derived from search response / `context` for BrowseView UX specifically */
+    const { showClearFiltersButton, countExternalSets, facets } = useMemo(function(){
+        const { filters: contextFilters = null, facets: contextFacets = [] } = context;
```
Comment (j1z0) [inline src/encoded/static/components/browse/BrowseView.js:394]: Codacy found an issue: ['context.filters' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/FileSearchView.js [hunk 4]
```diff
@@ -5,59 +5,38 @@ import PropTypes from 'prop-types';
...
+        columnExtensionMap,
+        currentAction
```
Comment (j1z0) [inline src/encoded/static/components/browse/FileSearchView.js:39]: Codacy found an issue: ['currentAction' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Clones removed
==============
+ src/encoded/static/components/browse/FileSearchView.js  -3
+ src/encoded/static/components/browse/PublicationSearchView.js  -2
+ src/encoded/static/components/browse/BrowseView.js  -1
+ src/encoded/static/components/browse/MicroscopySearchView.js  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004&bid=20599883)

File: src/encoded/static/components/browse/BrowseView.js (hunk 2 above)
Comment (j1z0) [inline src/encoded/static/components/browse/BrowseView.js:320]: Codacy found an issue: ['resetSelectedFiles' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/BrowseView.js [hunk 5]
```diff
@@ -537,40 +352,102 @@ function BrowseTableWithSelectedFilesCheckboxes(props){
...
```
Comment (j1z0) [inline src/encoded/static/components/browse/BrowseView.js:394]: Codacy found an issue: ['context.filters' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/FileSearchView.js (hunk 1 above)
Comment (j1z0) [inline src/encoded/static/components/browse/FileSearchView.js:37]: Codacy found an issue: ['resetSelectedFiles' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/FileSearchView.js (hunk 4 above)
Comment (j1z0) [inline src/encoded/static/components/browse/FileSearchView.js:39]: Codacy found an issue: ['currentAction' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Clones removed
==============
+ src/encoded/static/components/browse/PublicationSearchView.js  -2
+ src/encoded/static/components/browse/MicroscopySearchView.js  -3
+ src/encoded/static/components/browse/FileSearchView.js  -3
+ src/encoded/static/components/browse/BrowseView.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004&bid=20599883)

File: src/encoded/static/components/browse/BrowseView.js (hunk 2 above)
Comment (j1z0) [inline src/encoded/static/components/browse/BrowseView.js:320]: Codacy found an issue: ['resetSelectedFiles' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/BrowseView.js (hunk 5 above)
Comment (j1z0) [inline src/encoded/static/components/browse/BrowseView.js:394]: Codacy found an issue: ['context.filters' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/FileSearchView.js (hunk 1 above)
Comment (j1z0) [inline src/encoded/static/components/browse/FileSearchView.js:37]: Codacy found an issue: ['resetSelectedFiles' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/FileSearchView.js (hunk 4 above)
Comment (j1z0) [inline src/encoded/static/components/browse/FileSearchView.js:39]: Codacy found an issue: ['currentAction' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

**Review by utku-ozturk — APPROVED** [review]
We have tested both default and customized (File/Publication/MicroscopeConf) search views locally, works well.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Clones removed
==============
+ src/encoded/static/components/browse/FileSearchView.js  -2
+ src/encoded/static/components/browse/MicroscopySearchView.js  -3
+ src/encoded/static/components/browse/PublicationSearchView.js  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004&bid=20599883)

File: src/encoded/static/components/browse/FileSearchView.js [hunk 6]
```diff
@@ -5,59 +5,39 @@ import PropTypes from 'prop-types';
...
```
Comment (j1z0) [inline src/encoded/static/components/browse/FileSearchView.js:40]: Codacy found an issue: ['currentAction' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/FileSearchView.js [hunk 7]
```diff
@@ -5,59 +5,39 @@ import PropTypes from 'prop-types';
...
 function FileTableWithSelectedFilesCheckboxes(props){
     const {
         // Common high-level props from Redux, or App.js, or App.js > BodyElement:
         context, href, schemas, navigate: propNavigate,
         windowHeight, windowWidth, registerWindowOnScrollHandler,
         toggleFullScreen, isFullscreen, session,
-        facets,
-        selectedFiles, selectFile, unselectFile, resetSelectedFiles,
-        columnExtensionMap
+        selectedFiles, selectFile, unselectFile, resetSelectedFiles, selectedFilesUniqueCount,
```
Comment (j1z0) [inline src/encoded/static/components/browse/FileSearchView.js:38]: Codacy found an issue: ['resetSelectedFiles' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/BrowseView.js (hunk 2 above)
Comment (j1z0) [inline src/encoded/static/components/browse/BrowseView.js:320]: Codacy found an issue: ['resetSelectedFiles' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

File: src/encoded/static/components/browse/BrowseView.js (hunk 5 above)
Comment (j1z0) [inline src/encoded/static/components/browse/BrowseView.js:394]: Codacy found an issue: ['context.filters' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6342004)

**Pull request merged**